### DISCUSSION TITLE
Identifiable Array and TestStore

### DISCUSSION THREAD

**Discussion by ivangodfather** [pointfreeco/swift-composable-architecture#1074] [Q&A] [answered]
While doing the TCA tour videos I got stuck if I change the signature of the todo's to IdentifiedArrayOf (the suggested one).
Cannot assign to property: subscript is get-only

```
struct AppState: Equatable {
	var todos: IdentifiedArrayOf<Todo>
}

store.send(.todo(id: UUID(), action: .checkboxTapped)) {
        $0.todos[0].isComplete = true
}
```
<img width="1270" alt="Screenshot 2022-05-02 at 23 04 13" src="https://user-images.githubusercontent.com/1838117/166327461-4fa29b09-0fdd-432a-abbc-c319c61352df.png">

**Comment by stephencelis** [comment, marked as answer]
Hi @ivangodfather! Although many of the core principles remain the same from those episodes, there have been a bunch of API changes since recording.

The main issue here is that `IdentifiedArray` cannot be mutated via its index. Instead, you should be able to use the `id`-based subscript:

```swift
let id = UUID(uuidString: "00000000-0000-0000-0000-000000000000")!

store.send(.todo(id: id, action: .checkboxTapped) {
  $0.todos[id: id]!.isComplete = true
}
```
